functions/nodejs-apps.bash

```diff
@@ -342,3 +342,60 @@ zigbee2mqtt_setup() {
     whiptail --title "Operation successful" --msgbox "$installSuccessText" 15 80
   fi
 }
+
+## Function for installing a npm package for the JS Scripting Automation Add-On
+##
+##    jsscripting_npm_install(String packageName, String mode)
+##    Available values for mode: "update", "install". Defaults to "install".
+##
+jsscripting_npm_install() {
+  if [ "${1}" == "" ]; then echo "FAILED. Provide packageName."; return 1; fi
+
+  if ! node_is_installed || is_armv6l; then
+    echo -n "$(timestamp) [openHABian] Installing prerequsites for ${1} (NodeJS)... "
+    if cond_redirect nodejs_setup; then echo "OK"; else echo "FAILED"; return 1; fi
+  fi
+
+  if [ "${2}" == "update" ];
+  then
+    echo -n "$(timestamp) [openHABian] Updating ${1} for JS Scripting... "
+    if cond_redirect sudo -u "openhab" npm update --prefix "/etc/openhab/automation/js" "${1}"; then echo "OK"; else echo "FAILED (npm update)"; return 1; fi
+  else
+    echo -n "$(timestamp) [openHABian] Installing ${1} for JS Scripting... "
+    if ! cond_redirect sudo -u "openhab" mkdir -p /etc/openhab/automation/js; then echo "FAILED (mkdir /etc/openhab/automation/js)"; fi
+    if cond_redirect sudo -u "openhab" npm install --prefix "/etc/openhab/automation/js" "${1}"; then echo "OK"; else echo "FAILED (npm install)"; return 1; fi
+  fi 
+}
+
+## Function for checking for updates of a npm package for the JS Scripting Automation Add-On
+##
+##    jsscripting_npm_check(String packageName)
+##
+jsscripting_npm_check() {
+  if [ "${1}" == "" ]; then echo "FAILED. Provide packageName."; return 1; fi
+
+  # If directory of package doesn't exist, exit.
+  if [ ! -d "/etc/openhab/automation/js/node_modules/${1}" ]; then return 0; fi
+
+  if ! node_is_installed || is_armv6l; then
+    echo -n "$(timestamp) [openHABian] Installing prerequsites for ${1} for JS Scripting (NodeJS)... "
+    if cond_redirect nodejs_setup; then echo "OK"; else echo "FAILED"; return 1; fi
+  fi
+
+  introText="Additions, improvements or fixes were added to ${1} (npm package) for JS Scripting. Would you like to update now and benefit from them?\\nThe update might include breaking changes, please head over to the JS Scripting docs or to https://www.npmjs.com/package/${1}.\\n\\nUpdating is recommended."
```

Comment:
Mind the openHABian conventions (see CONTRIBUTING.md). Define consts like this as local at the beginning of the routine.